### ISSUE TITLE
metrics-*-* indices requires 1 replica in 8.12-SNAPSHOT

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#7376]
There have been a few test failures with `8.12.0-SNAPSHOT`:
- TestEntSearchAssociation
- TestKibanaAssociationWhenReferencedESDisappears
- TestKillKibanaPod
- TestKillKibanaDeployment
- TestKBStackMonitoring

due to an index being created with a replica, when the test cluster is only running with a single node.

```sh
.ds-metrics-fleet_server.agent_status-default-2023.12.05-000001 0 p STARTED      1   8.9kb   8.9kb 10.117.225.40 test-kill-kb-pod-d7x5-es-masterdata-0
.ds-metrics-fleet_server.agent_status-default-2023.12.05-000001 0 r UNASSIGNED
```

Index template:
```json
        {
            "name": "metrics",
            "index_template": {
                "index_patterns": [
                    "metrics-*-*"
                ],
                "composed_of": [
                    "metrics@mappings",
                    "data-streams@mappings",
                    "metrics@settings"
                ],
                "priority": 100,
                "version": 4,
                "_meta": {
                    "managed": true,
                    "description": "default metrics template installed by x-pack"
                },
                "data_stream": {
                    "hidden": false,
                    "allow_custom_routing": false,
                    "failure_store": false
                },
                "allow_auto_create": true,
                "deprecated": false
            }
        },
```

https://github.com/elastic/cloud-on-k8s/pull/7364 adds another node to these tests, until we figure out what needs to be reconfigured so that the index in question is created with `auto_expand_replicas` set to `0-1`, for example.

**Comment by naemono** [comment]
Adding `TestKibanaStandalone` to this list as well, and *note* it doesn't involve Fleet server or the Fleet plugin at all, but this fleet index is still being created with a replica.

**Comment by naemono** [comment]
This should be able to be resolved on our end, and the 2 node clusters removed when https://github.com/elastic/kibana/pull/173689 is fully merged and available in 8.12.0-snapshot.

**Comment by thbkrkr** [comment]
`TestTelemetry/ES_cluster_health_should_eventually_be_green` fails for the same reason in [build-399](https://buildkite.com/elastic/cloud-on-k8s-operator-nightly/builds/399#018c8bb5-fc52-4eac-a02f-f15b2e6819af):

`.ds-metrics-fleet_server.agent_status-default-2023.12.21-000001 0     r      UNASSIGNED`
